File: test-multilang/kotlin/UserService.kt
package com.example.service

import com.example.repository.UserRepository
import com.example.model.User

class UserService(private val userRepository: UserRepository) {
    
    fun findById(id: Long): User? {
        return userRepository.findById(id)
    }
    
    fun findAll(): List<User> {
        return userRepository.findAll()
    }
    
    fun createUser(name: String, email: String): User {
        val user = User(name = name, email = email)
        return userRepository.save(user)
    }
    
    fun deleteUser(id: Long): Boolean {
        return if (userRepository.existsById(id)) {
            userRepository.deleteById(id)
            true
        } else {
            false
        }
    }
    
    suspend fun findUserAsync(id: Long): User? {
        return userRepository.findByIdAsync(id)
    }
}

File: test-multilang/cpp/UserService.cpp
#include <iostream>
#include <vector>
#include <optional>
#include <string>

class UserService {
private:
    UserRepository* userRepository;

public:
    UserService(UserRepository* repo) : userRepository(repo) {}

    std::optional<User> findById(int id) {
        return userRepository->findById(id);
    }

    std::vector<User> findAll() {
        return userRepository->findAll();
    }

    User createUser(const std::string& name, const std::string& email) {
        User user(name, email);
        return userRepository->save(user);
    }

    bool deleteUser(int id) {
        if (userRepository->existsById(id)) {
            userRepository->deleteById(id);
            return true;
        }
        return false;
    }
};
